flipperdevices/flipperzero-firmware
# [FL-3322] Infrared: respect carrier frequency and duty cycle settings

# What's new

- Take carrier frequency and duty cycle into account for saved RAW signals.
- Update infrared_worker comments.

# Fixes
- #2663 
- #2632

# Verification 

1. Build firmware with Infrared TX debugging enabled: 
```sh
./fbt --extra-define=INFRARED_TX_DEBUG flash
```
2. Go to `Infrared -> Saved Remotes` and load [the test file](https://github.com/flipperdevices/flipperzero-firmware/files/11507676/ARRIS_test.zip).
3. Attach the oscilloscope probe to `PA7` (Pin 2).
4. Send the signal and ensure that the carrier frequency and duty cycle match those specified in the file.

# Checklist (For Reviewer)
